### ISSUE TITLE
failing on recent openssl in debian testing/unstable

### ISSUE THREAD

**Issue by avsm** [janestreet/async_ssl#25]
```
#=== ERROR while compiling async_ssl.v0.9.114.40+02 ===========================#
# command              jbuilder build --only-packages async_ssl --root . -j 1 @install
# path                 /home/opam/.opam/ocaml-base-compiler.4.03.0/.opam-switch/build/async_ssl.v0.9.114.40+02
# exit-code            1
# env-file             /home/opam/.opam/log/async_ssl-21-337a38.env
# output-file          /home/opam/.opam/log/async_ssl-21-337a38.out
### output ###
# [...]
#         bash src/ffi_generated.ml
#         bash src/ffi_generated_stubs.c
#         bash stubgen/ffi_ml_types_stubgen.c
#          ppx src/ffi_generated.pp.ml
#       ocamlc src/ffi_generated_stubs.o
#         bash stubgen/ffi_ml_types_stubgen.exe (exit 1)
# (cd _build/default/stubgen && /usr/bin/bash -e -u -o pipefail -c 'gcc  -std=gnu99 -O2 -fno-strict-aliasing -fwrapv -Wall -D_FILE_OFFSET_BITS=64 -D_REENTRANT -fPIC ffi_ml_types_stubgen.c $(< ../bindings/openssl-ccopt) $(< ../bindings/openssl-cclib) -I `dirname /home/opam/.opam/ocaml-base-compiler.4.03.0/lib/ctypes/ctypes_cstubs_internals.h` -I /home/opam/.opam/ocaml-base-compiler.4.03.0/lib/oc[...]
# ffi_ml_types_stubgen.c: In function 'main':
# ffi_ml_types_stubgen.c:39:49: error: 'SSL_CTRL_OPTIONS' undeclared (first use in this function)
#       enum { check_SSL_CTRL_OPTIONS_const = (int)SSL_CTRL_OPTIONS };
#                                                  ^~~~~~~~~~~~~~~~
# ffi_ml_types_stubgen.c:39:49: note: each undeclared identifier is reported only once for each function it appears in

```

full log: https://ci.ocaml.io/log/saved/docker-build-468846e644d3d10891fb511649c0e0e8/dd04d1de84208d1e562f578dc16235c9e94dae50

**Comment by rgrinberg** [comment]
This is also failing for me on Archlinux.

**Comment by vbmithr** [comment]
See #26

**Comment by tiash** [comment]
I believe the SSL_CTRL_OPTIONS is an OpenSSL internal that changed sometime between 1.0 and 1.1.

SSL_CTX_ctrl and SSL_CTRL_OPTIONS are both undocumented internal functions that we shouldn't be using. I've changed the code to use the correct macros/function instead in branch [openssl-version-portibility-fixes](https://github.com/janestreet/async_ssl/tree/openssl-version-portibility-fixes).

**Comment by vbmithr** [comment]
Le 30/05/2017 à 13:38, Matthias Horn a écrit :
> I believe the SSL_CTRL_OPTIONS is an OpenSSL internal that changed
> sometime between 1.0 and 1.1.
> 
> SSL_CTX_ctrl and SSL_CTRL_OPTIONS are both undocumented internal
> functions that we shouldn't be using. I've changed the code to use the
> correct macros/function instead in branch
> openssl-version-portibility-fixes
> <https://github.com/janestreet/async_ssl/tree/openssl-version-portibility-fixes>.
> 

Works great now, thanks!

-- 
Vincent

**Comment by avsm** [comment]
@dim could we get a release cut with this? It's also blocking automated testing of mirage/ocaml-conduit

**Comment by avsm** [comment]
(I can test it on various Linux distros via a travis file if you want to activate that on this repo)

**Comment by avsm** [comment]
see https://travis-ci.org/mirage/ocaml-conduit/builds/239723876?utm_source=github_status&utm_medium=notification for a test matrix of failure

**Comment by diml** [comment]
@tiash can you backport the fix in the `v0.9` branch? I'll do a release after that

**Comment by trefis** [comment]
The changes have been merged into master and I backported them on the v0.9 branch.
@diml I leave it to you to close the issue once the release is done.

**Comment by avsm** [comment]
Thanks! @diml this came up as a revdep failure in https://github.com/ocaml/opam-repository/pull/9892, so a release would be appreciated

**Comment by trefis** [comment]
I'll try and do one *soon*.

**Comment by trefis** [comment]
https://github.com/ocaml/opam-repository/pull/10008

**Comment by hhugo** [comment]
@avsm, Is this resolved ?

**Comment by hhugo** [comment]
Probably solved. Reopen if needed.
